src/pybamm/citations.py

```diff
@@ -68,7 +68,9 @@ def read_citations(self):
         """
         try:
             parse_file = import_optional_dependency("pybtex.database", "parse_file")
-            citations_file = os.path.join(pybamm.root_dir(), "pybamm", "CITATIONS.bib")
+            citations_file = os.path.join(
+                pybamm.root_dir(), "src", "pybamm", "CITATIONS.bib"
+            )
```

Comment:
I figured out the problem with the failing RTD builds, and it's coming from these lines:

Note the location of the `CITATIONS.bib` file in the logs:

```
FileNotFoundError: [Errno 2] No such file or directory: '/home/docs/checkouts/readthedocs.org/user_builds/pybamm/envs/4311/lib/python3.12/src/pybamm/CITATIONS.bib'
```

i.e., the path is `lib/python3.12/src/pybamm/CITATIONS.bib`, which is obviously incorrect – it's not going to exist. The correct path to check should be `lib/python3.12/site-packages/pybamm/CITATIONS.bib`. 

This means that the output from `pybamm.root_dir()` was always incorrect in the first place, and the problem has just been exposed when migrating to an `src` layout. The reason why this doesn't fail outside of the RTD builds is because we use an editable installation with `nox` in the PR checks (the `-e` command-line flag), while the RTD builds don't install in editable mode. You can verify for yourself and reproduce the same error locally:

``` bash
# works
pip install -e .
python -c "import pybamm; print(pybamm.root_dir())"
# fails
pip install .
python -c "import pybamm; print(pybamm.root_dir())"
```

I notice that `CITATIONS.bib` exists in the same folder as `citations.py`. Maybe you could modify these lines to fix this?

Reply:
Thanks for the suggestions. I was able to reproduce this locally.